src/hooks/useOrders.ts

```diff
@@ -27,19 +27,62 @@ function filterDeletedOrders(orders: AuctionElement[]): AuctionElement[] {
   )
 }
 
-export function useOrders(): AuctionElement[] {
+interface Result {
+  orders: AuctionElement[]
+  forceOrdersRefresh: () => void
+}
+
+export function useOrders(): Result {
   const { userAddress, networkId, blockNumber } = useWalletConnection()
   const [orders, setOrders] = useSafeState<AuctionElement[]>([])
+  const [lastSeenOrderId, setLastSeenOrderId] = useSafeState<number>(-1)
+  const [offset, setOffset] = useSafeState<number | undefined>(undefined)
+
+  /**
+   * Forces the immediate refresh of orders
+   */
+  function forceOrdersRefresh(): void {
+    // Setting `offset` to 0 triggers a new query from the start
+    setOffset(0)
+  }
+
+  useEffect(() => {
+    // whenever there's a state change (new block, network /address change),
+    // set `offset` to trigger a new query
+    setOffset(lastSeenOrderId + 1)
+  }, [blockNumber, networkId, userAddress])
```

Comment:
Change in `networkId, userAddress` should refetch all orders as all orders are network|user specific

Reply:
You are 100% correct. After submitting this PR I started working on this. Unfortunately, I haven't been able to make it work it.
It behaves weird when switching network/account for some reason. Still trying to figure it out.

Reply:
Got to a somewhat working solution with a bunch of `useEffect`s and duplicated block, network and address. Ugly.

Any suggestions to make it better?